csrc/id_model/id_model.cpp

```diff
@@ -1439,4 +1444,29 @@ Val* IdModel::getLoopIndexVariable(
   return getLoopIndexVariable(loop_group, circular_buffer_loop_stage);
 }
 
+ValGraph buildPermissiveResizeGraph(IdModel& id_model) {
```

Comment:
It would be clearer this way rather than taking the non-const reference:

```suggestion
ValGraph buildPermissiveResizeGraph(const ValGraph& permissive_graph) {
```